File: api/src/main/java/io/github/manishdait/aquila/ticket/Ticket.java
package io.github.manishdait.aquila.ticket;

import java.time.Instant;
import java.util.List;

import io.github.manishdait.aquila.project.Project;
import io.github.manishdait.aquila.users.User;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToMany;
import jakarta.persistence.ManyToOne;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Builder
@Entity
public class Ticket {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String title;
    private String description;
    private Instant createAt;
    private Instant updatedAt;
    @Enumerated(EnumType.STRING)
    private Status status;
    @Enumerated(EnumType.STRING)
    private Priority priority;
    @ManyToOne
    @JoinColumn(name = "reported_by")
    private User reportedBy;
    @ManyToMany
    private List<User> assignees;
    @ManyToOne
    @JoinColumn(name = "project")
    private Project project;
}

File: client/aquila-ui/src/app/model/ticket.ts
import { UserResponse } from "./user";

export interface TicketRequest {
    title: string,
    description: string,
    priority: string,
    assignees: string[],
    projectId: number
}

export interface TicketResponse {
    readonly id: number,
    title: string,
    description: string,
    createdAt: Date | null,
    updatedAt: Date | null,
    priority: string,
    status: string,
    reportedBy: UserResponse | null,
    assignees: UserResponse[],
    readonly projectId: number,
    projectCode: string,
    commentCount: number
}